src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -970,7 +971,7 @@ private bool ImportIntrinsicCall(MethodDesc method)
             return false;
         }
 
-        private void HandleCall(MethodDesc callee, ILOpcode opcode = ILOpcode.call)
+        private void HandleCall(MethodDesc callee, ILOpcode opcode = ILOpcode.call, TypeDesc forcedReturnType = null)
```

Comment:
I couldn't find the case where you needed to force a return type. Would it be possible to cast? I'm a little worried that we could end up with conflicting signatures.

Reply:
NewObject and NewArray both end up putting a LoadExpression on the stack with a type of Object/Array instead of the real type. I'm not sure anything is currently running into issues without this but once we start doing castclass stuff I think having the correct type on the stack is going to start to matter.